### PR TITLE
Include map length in diagnostics

### PR DESCRIPTION
**Pull request by Manishearth**
If this doesn't match with the size it's more likely for anything caught
t be hardware corruption. If it does the situation is more interesting.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18977)

### CODE REVIEW

**Comment by Gankro** [comment]
This definitely merge conflicts with the hash patch.

**Comment by Manishearth** [comment]
Oh, that didn't land yet. Rebased. Can we r+ this with higher priority so both land?

File: components/hashglobe/src/diagnostic.rs [hunk 1]
```diff
@@ -62,17 +62,21 @@ impl<K: Hash + Eq, V, S: BuildHasher> DiagnosticHashMap<K, V, S>
...
                 continue;
             }
+
             count += 1;
-            bad_canary = Some((*canary_ref, canary_ref, position));
+            bad_canary = Some((h, *canary_ref, canary_ref, position));
         }
         if let Some(c) = bad_canary {
-            self.report_corruption(c.0, c.1, c.2, count);
+            self.report_corruption(c.0, c.1, c.2, c.3, count, self.map.diagnostic_count_hashes());
```
Comment (Gankro) [inline components/hashglobe/src/diagnostic.rs]: This can just be computed in report_bad_corruption.

**Review by Gankra — COMMENTED** [review]
r=me with one change

dunno how queues or whatever should work, y'all can figure it out.

**Pull request merged**